src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1523,6 +1731,206 @@ def forward(
         )
 
 
+@add_start_docstrings(
+    """
+    BLIP-2 Text Model with a projection layer on top (a linear layer on top of the pooled output).
+    """,
+    BLIP_2_START_DOCSTRING,
+)
+class Blip2TextModelWithProjection(Blip2PreTrainedModel):
```

Comment:
Curious what the purpose is of this class (along with `Blip2VisionModelWithProjection`)?

Did the authors release weights for these (don't see them being used in the conversion script)?